base/compiler/typeinfer.jl

```diff
@@ -194,19 +94,7 @@ end  # module Timings
 If set to `true`, record per-method-instance timings within type inference in the Compiler.
 """
 __set_measure_typeinf(onoff::Bool) = __measure_typeinf__[] = onoff
-const __measure_typeinf__ = fill(false)
-
-# Wrapper around `_typeinf` that optionally records the exclusive time for each invocation.
-function typeinf(interp::AbstractInterpreter, frame::InferenceState)
-    if __measure_typeinf__[]
-        Timings.enter_new_timer(frame)
-        v = _typeinf(interp, frame)
-        Timings.exit_current_timer(frame)
-        return v
-    else
-        return _typeinf(interp, frame)
-    end
-end
+const __measure_typeinf__ = RefValue{Bool}(false)
```

Comment:
@vtjnash: How does this PR impact `@snoopi_deep`? It looks like this was disabled.

Will there be a new way to get instrumented inference profiles, showing per-frame inference times?

CC: @IanButterworth

Reply:
I wasn't certain how to maintain the old code, since it was working with incomplete and inaccurate information available. Now all required information is available in `function typeinf(interp::AbstractInterpreter, frame::InferenceState)`, so this could be reimplemented there in some form.